source/install/upgrade_from_mongodb.rst

```diff
@@ -4,6 +4,11 @@
 Upgrading from |mongodb-ce|
 ================================================================================
 
+An in-place upgrade is done with existing data in the server.  Generally
+speaking, this is stopping the server, removing the old packages, installing the
+new server and starting it with the same data files.  While an in-place upgrade
+may not be suitable for high-complexity environments, it should work in most
+cases.
```

Comment:
Regarding _"While an in-place upgrade_
_may not be suitable for high-complexity environments, it should work in most_
_cases."_

An in-place upgrade is suitable in all cases I think except the following:
* ephemeral hosts are used, ones that get new IP addresses and/or hostnames if they are restarted, and/or
* disk storage is ephemeral

The exceptions above are only situations you would get into with container-based systems or some cloud virtual machine services.

Classic on-prem servers, or VMs that are used like classic servers, can (and should i.m.o.) use in-place updates even if they are high-complexity in other ways.

Reply:
So instead of "may not be suitable for high-complexity environments" I suggest "will not be suitable for environments that use ephemeral storage or host addresses"